### PR TITLE
fix: `scripts/build_dependencies.sh` no longer fails on linux

### PR DESCRIPTION
**Pull request by tsansbury-rmn**

### CODE REVIEW

File: scripts/build_arrow_source.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ ARROW_DEPS_BUILD_DIR=$DEPENDENCY_DIR/arrow_deps
 ARROW_CMAKE_BUILD_DIR=$ARROW_SOURCE_DIR/cpp/cmake-build
 
 GIT_REPO="https://github.com/apache/arrow.git"
-CLONE_CMD="git clone -b master $GIT_REPO $ARROW_SOURCE_DIR"
+CLONE_CMD="git clone $GIT_REPO $ARROW_SOURCE_DIR"
```
Comment (tsansbury-rmn) [inline scripts/build_arrow_source.sh:29]: The arrow project (https://github.com/apache/arrow) renamed their `master` branch to `main`.  I suspect the default branch is fine for the clone target given subsequent steps will checkout a versioned tag.

File: scripts/build_azuresdk.sh [hunk 2]
```diff
@@ -74,18 +74,11 @@ azure_configure_opts+=(
...
-if [[ "$ARCH" == "aarch64" || "$ARCH" == "arm64" ]]; then
-  azure_configure_opts+=(
-      "-DBUILD_TESTS=false"
-  )
-else
-  azure_configure_opts+=(
-      "-DBUILD_TESTS=true"
-  )
-fi
+azure_configure_opts+=(
```
Comment (tsansbury-rmn) [inline scripts/build_azuresdk.sh:79]: Azure test failed compilation due to an older version of Catch was failing for both Linux x86_64 and aarch64.

**Comment by sfc-gh-ext-simba-hx** [comment]
Hi @tsansbury-rmn ,
The fixes here is merged in https://github.com/snowflakedb/libsnowflakeclient/pull/501

**Comment by tsansbury-rmn** [comment]
Wonderful.  Thank you.

**Pull request closed without merging**